### ISSUE TITLE
Limitation: auto-split only detected on logout/login (steam install)

### ISSUE THREAD

**Issue by jeske** [brandondong/POE-LiveSplit-Component#11]
TL;DR - In my testing (steam), PoE only writes zone entry log lines on login/logout, and so splits are only auto-detected on login/logout. This may not matter for seasoned speed-runners, but for players new to speed running PoE, logout/login is not "normal" and doing it all the time is not common.

If the non-steam version has a different log behavior, this could explain why steam users are having trouble.

---

### More Details

In my first testing auto-splitting "doesn't seem to work", because when I walk into Lioneye's Watch, there is no split detected.

However, if I logout and log back in, it auto-detects the split.

This seems like a limitation of Path of Exile (steam), as when I just walk into LionEye's watch, it shows "You have entered Lioneye's Watch." in the on screen GUI, but there is no entry in the logfile. 

![image](https://user-images.githubusercontent.com/15093/112687102-31ce9d80-8e34-11eb-9d9e-be420c9ce889.png)

![image](https://user-images.githubusercontent.com/15093/112687118-398e4200-8e34-11eb-9a00-bdc3494d4cfe.png)

**Comment by brandondong** [comment]
Thanks for the investigation. That is very interesting. I use the standalone client and these are my logs shortly after I opened the game, created a new character, and made my way to Lioneye's Watch.
```
2021/03/26 21:17:04 39819968 16 [INFO Client 9168] [STARTUP] Bundle Tiny in 6.2e-05 seconds
2021/03/26 21:17:12 39828421 bf [INFO Client 9168] Async connecting to sjc01.login.pathofexile.com:20481
2021/03/26 21:17:12 39828484 e6 [INFO Client 9168] Connected to sjc01.login.pathofexile.com in 31ms.
2021/03/26 21:17:40 39855671 eaf [DEBUG Client 9168] Got Instance Details from login server
2021/03/26 21:17:40 39855687 ecc [INFO Client 9168] Just before calling client instance session
2021/03/26 21:17:40 39855703 10e [INFO Client 9168] Connecting to instance server at 216.52.30.147:6112
2021/03/26 21:17:40 39855812 1c9 [DEBUG Client 9168] Connect time to instance server was 93ms
2021/03/26 21:17:43 39859406 bb3 [INFO Client 9168] : You have entered The Twilight Strand.
2021/03/26 21:18:52 39927562 bb3 [INFO Client 9168] : test_log_entries (Witch) is now level 2
2021/03/26 21:18:57 39932890 eaf [DEBUG Client 9168] Got Instance Details from login server
2021/03/26 21:18:57 39932906 ecc [INFO Client 9168] Just before calling client instance session
2021/03/26 21:18:57 39932906 10e [INFO Client 9168] Connecting to instance server at 216.52.30.141:6112
2021/03/26 21:18:57 39932953 1c9 [DEBUG Client 9168] Connect time to instance server was 47ms
2021/03/26 21:18:58 39933843 bb3 [INFO Client 9168] : You have entered Lioneye's Watch.
```
It appears your log entries are missing the Lioneye's Watch entrance log as well as the level up notification. This suggests that this tool is completely useless for Steam users (autosplitting by zone/level, load removal, and lab speedrunning will all fail). I did test the tool with the Steam client some ages ago and there were no such logging discrepancies then. When I get the time, I'll have to redownload that client and verify the issue.

**Comment by richardcocks** [comment]
I'm not sure how helpful this is, but as a workaround or something to debug further, I've found auto-splitting has been unproblematic for me on steam as long as I don't have the Lioneye's watch split included. With my first split as The Coast it has worked fine compared to LW first which has been unreliable.

**Comment by brandondong** [comment]
I now have the Steam version installed and cannot reproduce the missing log entries. Here they are running the same test in my above comment.
```
021/03/29 21:45:57 46189875 16 [INFO Client 10032] [STARTUP] Bundle Tiny in 7.6e-05 seconds
2021/03/29 21:46:05 46197812 bf [INFO Client 10032] Async connecting to sjc01.login.pathofexile.com:20481
2021/03/29 21:46:06 46198750 e6 [INFO Client 10032] Connected to sjc01.login.pathofexile.com in 297ms.
2021/03/29 21:46:56 46249390 eaf [DEBUG Client 10032] Got Instance Details from login server
2021/03/29 21:46:56 46249406 ecc [INFO Client 10032] Just before calling client instance session
2021/03/29 21:46:56 46249421 10e [INFO Client 10032] Connecting to instance server at 63.251.253.85:6112
2021/03/29 21:46:57 46249718 1c9 [DEBUG Client 10032] Connect time to instance server was 297ms
2021/03/29 21:46:59 46252437 bb3 [INFO Client 10032] : You have entered The Twilight Strand.
2021/03/29 21:48:20 46333453 bb3 [INFO Client 10032] : test_log_entries_steam (Witch) is now level 2
2021/03/29 21:48:26 46338765 eaf [DEBUG Client 10032] Got Instance Details from login server
2021/03/29 21:48:26 46338781 ecc [INFO Client 10032] Just before calling client instance session
2021/03/29 21:48:26 46338781 10e [INFO Client 10032] Connecting to instance server at 173.237.53.244:6112
2021/03/29 21:48:26 46338953 1c9 [DEBUG Client 10032] Connect time to instance server was 156ms
2021/03/29 21:48:27 46339687 bb3 [INFO Client 10032] : You have entered Lioneye's Watch.
2021/03/29 21:49:28 46400875 eaf [DEBUG Client 10032] Got Instance Details from login server
2021/03/29 21:49:28 46400875 ecc [INFO Client 10032] Just before calling client instance session
2021/03/29 21:49:28 46400875 10e [INFO Client 10032] Connecting to instance server at 216.52.30.152:6112
2021/03/29 21:49:28 46400921 1c9 [DEBUG Client 10032] Connect time to instance server was 31ms
2021/03/29 21:49:30 46402750 bb3 [INFO Client 10032] : You have entered The Coast.
```
I have both the Lioneye's Watch entrance log as well as the level up notification. I have tried other zone orders/logins and the logs are written exactly as one would expect.
@jeske Just to make sure I understand completely what's going on, if you next enter The Coast in your provided screenshot, would that entry also be missing? And the same for reaching level 3?

Just for sanity sake, are you using C:\Program Files (x86)\Steam\steamapps\common\Path of Exile\logs\Client.txt? Are the missing log entries reproducible when LiveSplit is not open? Any other possibly notable third party tools running that may be reading the log file?

@richardcocks There should be no special logic in this application for splitting on Lioneye's Watch. If this ever happens to you in the future, it would be greatly appreciated if you could attach your Client.txt logs. I have tried myself several times with splitting and not splitting on Lioneye's Watch and I have consistently seen no issues.

**Comment by alexanderkitsou** [comment]
Small bump, I experienced the same thing, I dug through the posts without finding a solution but this posted seemed to have the same issue I did.   
After half an hour of trying different things and reading through logs an idea popped into my mind.  
What if the logs message "You have entered zone" is written only when Local chat is enabled.  
Toggled it on and it fixed my issue, and if I see your screenshot correctly it SHOULD fix yours as well ( you have local chat disabled ).

**Comment by brandondong** [comment]
> Small bump, I experienced the same thing, I dug through the posts without finding a solution but this posted seemed to have the same issue I did. After half an hour of trying different things and reading through logs an idea popped into my mind. What if the logs message "You have entered zone" is written only when Local chat is enabled. Toggled it on and it fixed my issue, and if I see your screenshot correctly it SHOULD fix yours as well ( you have local chat disabled ).

Nice find! I updated the README to mention this requirement.
